src/publish.proj

```diff
@@ -8,6 +8,15 @@
     <PublishPattern Condition="'$(PublishPattern)' == ''">$(PackageOutputRoot)**\*.nupkg</PublishPattern>
   </PropertyGroup>
 
+  <ItemGroup>
+    <_PackagesToPublish Include="$(PublishPattern)" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <!-- only try to publish if there are .nupkgs produced -->
+    <PublishBuildDependsOn Condition="'@(_PackagesToPublish)' != ''">CreateContainerName;UploadToAzure</PublishBuildDependsOn>
```

Comment:
I don't think this works.  Properties outside a target are evaluated before items.  https://msdn.microsoft.com/en-us/library/dd997067.aspx#Anchor_2

I would have expected this to error with
```
error MSB4099: A reference to an item list at position n is not allowed in this condition
```